Api/V8/Service/ModuleService.php

```diff
@@ -111,23 +115,88 @@ public function getRecords(GetModulesParams $params, Request $request)
         // negative numbers are validated in params
         $offset = $number !== 0 ? ($number - 1) * $size : $number;
         $realRowCount = $this->beanManager->countRecords($module, $where);
-        $limit = $size === BeanManager::DEFAULT_ALL_RECORDS ? BeanManager::DEFAULT_LIMIT : $size;
+        $limit = $size === BeanManager::DEFAULT_ALL_RECORDS
+            ? BeanManager::DEFAULT_LIMIT : $size;
         $deleted = $params->getDeleted();
 
         if (empty($fields)) {
             $fields = $this->beanManager->getDefaultFields($bean);
         }
 
-        $beanListResponse = $this->beanManager->getList($module)
-            ->orderBy($orderBy)
-            ->where($where)
-            ->offset($offset)
-            ->limit($limit)
-            ->max($size)
-            ->deleted($deleted)
-            ->fields($this->beanManager->filterAcceptanceFields($bean, $fields))
-            ->fetch();
+        // Detect if bean has email field
+        if ((property_exists($bean, 'email1')
+                && strpos($where, 'email1') !== false)
+            || (property_exists($bean, 'email2')
+                && strpos($where, 'email2') !== false)
+        ) {
+            $selectedFields = strtolower($module) . '.' . implode(','
+                    . strtolower($module[0]) . '.', $fields);
+            $selectedModule = strtolower($module);
+
+            // Selects Module or COUNT(*) and will add one to the query.
+            $idSelect = 'SELECT ' . $selectedModule . '.id ';
+            $countSelect = 'SELECT COUNT(*) AS cnt ';
+
+            $quotedCountSelect = $db->quote($countSelect);
+
+            // Email where clause
+            $fromQuery
+                = 'FROM email_addresses join email_addr_bean_rel on email_addresses.id = email_addr_bean_rel.email_address_id join '
+                . $selectedModule . ' on ' . $selectedModule
+                . '.id = email_addr_bean_rel.bean_id ';
+            $modifiedWhere = str_replace('accounts.email1',
+                'email_addresses.email_address', $where);
+            $where = (string)$modifiedWhere;
```

Comment:
Return value of `str_replace` is string|array, straight casting an array to a string produces a notice.  
This code will either do nothing, or produce a notice and cast the array to a string containing "Array".